updater_test.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/go-logr/logr"
+	logrTesting "github.com/go-logr/logr/testing"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+// mockSidecarHandler mocks minimal interface of sidecar handler that updater needs, rendering strings and
+// returning sorted list of objects that assembleConf() method was called for.
+type mockSidecarHandler struct {
+	assembleConfResults []string
+}
+
+func (h *mockSidecarHandler) assembleConf(pod *corev1.Pod, className string) (string, error) {
+	val := fmt.Sprintf("%s/%s/%s", pod.Namespace, pod.Name, className)
```

Comment:
is this a path name or just a `/` separated string?

Reply:
This is used to verify which secrets the `assembleConf` was called for. I changed separator to a `.` and added comment to make that clear.